build/cloudbuild-tf-apply.yaml

```diff
@@ -14,31 +14,51 @@
 
 timeout: 1200s
 steps:
-- name: gcr.io/$PROJECT_ID/terraform
+- id: "set SA"
+  name: gcr.io/${PROJECT_ID}/terraform
   entrypoint: /bin/bash
   args:
   - -c
   - |
     gcloud config set auth/impersonate_service_account ${_TF_SA_EMAIL}
```

Comment:
@rjerrems do we need this step?
based on over conversation on chat for `1-org` and onwards the jenkins/cloudbuild SA will have impersonate the `_TF_SA_EMAIL` via the provider alias right?

Reply:
@bharathkkb I think this step necessary for the `terraform init` regarding the GCS backend

Reply:
This is required for gcloud commands when using impersonation, all terraform things will be covered by provider config

Reply:
this doesn't affect the `terraform init` it is mainly for modules that use gcloud commands

Reply:
@rjerrems dont we pass in the impersonate option when we call the gcloud module as well?

Reply:
yup for the project factory. In the past, there were a number of modules running raw gcloud commands (network, GKE etc)  - this might be reduced now, but I don't see great harm setting this for the build?

Reply:
@rjerrems totally agreed, I was just wondering. Should we also just add this as a step in Jenkinsfile for parity?

Reply:
Yes I think we should, the steps should be as closely aligned as possible

Reply:
Added comment to Jenkinsfile PR for tracking.